pontoon/translations/views.py

```diff
@@ -309,6 +319,16 @@ def approve_translation(request):
         plural_form=translation.plural_form,
     )
 
+    # Update Review Master Badge stats
+    badge_thresholds = [5, 50, 250, 1000]
+    for level, threshold in enumerate(badge_thresholds, start=1):
+        if (
+            user.profile.review_master_level < level
+            and len(user.badges_reviewed_translations) >= threshold
```

Comment:
These two lines will result in 12 additional DB queries, which is a lot for each approve / reject operation, because DB queries are slow.

I'm thinking if it'll be better to just store translation / review / promotion counts in the DB for each user, and then use these counts here to check if we need to send a "badge awarded" notification to the user and again in the Profile page to determine which badge to show.

By doing that we wouldn't need to maintain the `badges_inception` date.

Thoughts?

Reply:
It turns out checking translation/review counts is fast and can be done as part of saving/reviewing translations. The same also applies for promotions, of course, which happens much less often.

```python
>>> from django.utils import timezone
>>> from datetime import datetime
>>> from pontoon.base.models import User
>>> from pontoon.actionlog.models import ActionLog
>>>
>>> user = User.objects.get(email="matjaz.horvat@gmail.com")
>>> ActionLog.objects.filter(
    performed_by=user,
    action_type="translation:created",
    created_at__gte=timezone.make_aware(datetime(2024,10,1))
).count()
```

So I suggest we complerely avoid writing to the DB and don't use fields introduced in https://github.com/mozilla/pontoon/pull/3397.